### DISCUSSION TITLE
Install Guide for Outline in Proxmox VE Helper-Scripts

### DISCUSSION THREAD

**Discussion by burgerga** [community-scripts/ProxmoxVE#556] [Guides]
> [!IMPORTANT]
> For a functional Outline you need
> * SSL: Either via reverse proxy (e.g., [Nginx Proxy Manager](https://community-scripts.github.io/ProxmoxVE/scripts?id=nginxproxymanager)) or setting a certificate manually (see the [Outline SSL documentation](https://docs.getoutline.com/s/hosting/doc/ssl-pzk7WO8d1n)). Here we assume you are using a reverse proxy.
> * Authentication: You should configure one of the Authentication methods listed in the [Outline Authentication documentation](https://docs.getoutline.com/s/hosting/doc/authentication-7ViKRmRY5o)

This guide will explain how to fully install a Outline LXC using [Proxmox VE Helper-Scripts](https://community-scripts.github.io/ProxmoxVE). 

1. Start by installing an Outline LXC using [Proxmox VE Helper-Scripts](https://community-scripts.github.io/ProxmoxVE/):
    ```bash
    bash -c "$(wget -qLO - https://github.com/community-scripts/ProxmoxVE/raw/main/vm/outline.sh)"
    ```
    (this will take some time)
2. After installation, enter the shell of your Outline LXC
3. Go to the directory where Outline is installed: 
    ```bash 
    cd /opt/outline
    ```
4. Edit `/opt/outline/.env` (e.g., with `nano .env`):
    - Add a value for `URL`, this should be a "fully qualified, publicly accessible URL" (although you can limit access to LAN only using your reverse proxy)
    - Add an authentication method, see the [Outline Authentication documentation](https://docs.getoutline.com/s/hosting/doc/authentication-7ViKRmRY5o) 
    - (Optional) Add SMTP configuration to enable email (e.g. for notifications and invitation links).
> [!NOTE]
> Have a look at `/opt/outline/.env.sample` for example configuration
> **NB** Although it says "at least ONE OF EITHER Google, Slack, or Microsoft is required for a working installation" this is not the case, and any of the supported authentication methods should work.
5. Generate the database for outline with
```bash
yarn sequelize db:create
```
6. Restart Outline with
```bash
systemctl restart outline
```
7. If you pointed your `URL` to http://<OUTLINE_LXC_IP>:3000 in your reverse proxy, you should now be able to login by going to `URL`!

**Comment by Aloe-recite** [comment]
If I may, I would add that valid authentication methods are, as of now, Google Workspaces, MS Azure or Slack. It took me a few trials to understand that free Gmail is not valid and Github integration is not for OAuth but solely for merging PRs when writing your notes. Their documentation is not as clear as I'd like, and the mechanism is not exaclty straightforward. I ended up being stuck since I didn't have any of the mentioned providers, and ended up riesumating an old slack accunt tied to an old project, which is btw less than ideal and I'll probably activate a fresh ne account. Maintaining this additional information (likely to change in the future maybe) in Community-Scripts is less than ideal, but I would weigh it against the risk of many new issues/discussions from people not understaing how to set it up.
Just my 2c; sorry for butting in. BTW outline is working great with your beta script and is very light on resources, much more than anticipated, although AppFlowy is starting to tempt me with its AI integration... 😄

**Reply by burgerga** [reply]
Thanks for your note! The maintainers also had issues setting it up, so in the current state it will not be added as community-script.

I'm using OIDC, which is not too bad to set up as self-hosted service (e.g., using Authelia/Authentik/KeyCloak (probably also in that order)). Did you try https://docs.getoutline.com/s/hosting/doc/google-hOuvtCmTqQ for GMail? Because I added Google OAuth to Authentik using this approach, and that totally worked with a free GMail address (although Google's documentation is also terrible, so it took some tries).

Edit: Could work without Google Workspace: Just pick External instead of Internal (see for example the Authentik instructions: https://docs.goauthentik.io/docs/users-sources/sources/social-logins/google/)

**Reply by Aloe-recite** [reply]
Yes the page you linked says “Using Google as an OAuth provider is only supported if you are using Google Workspace” and unfortunately it’s true. I’m not using Authentik for the time being, we will see in the next future. It’s another thing to understand, setup and maintain for me but for Outline seems like the best bet. Setting up Slack was very easy too, it just took me a while to first try Google then GitHub all the while not realising it wasn’t supported. Unfortunately the selection of providers is too small but the it’s easy and you always have the OIDC route.

**Comment by hb30315** [comment]
I would like to ask if there is a detailed setting method, because when I tried to set up Authentik and outline recently, I found that  authentication failed. I referred to the video https://youtu.be/ywQVe9ikcVI?si=3jucNVaAXJ90UTQ5
